### PR TITLE
Add immediate dispatcher provider and apply it to MessageComposerController.kt

### LINKED ISSUE #3349: Investigate MessageComposer state & cursor update issues
### Context & proposal

We currently have problems where the state updates in the MessageComposer/TextField don't update the cursor correctly or the text in case of deleting items.

This is due to non-main-thread collection and updates of flows. Apparently the TextField state needs to be fully blocking and synchronous, otherwise it can have race conditions in state updates and show incorrect state.

We should try investigating if using a different dispatcher or way to update the `text` state of the Composer improves this handling.

### Acceptance criteria

To accept this task we need to complete the following items:

- [x] Try to update to `Dispatchers.Main.immediate` to force no dispatching events on text changes.
- [x] Try using `Dispatcher.Unconfined` to keep in the same thread without dispatching events.
- [x] Analyze if we can update the state in another, synchronous way.

### PR DESCRIPTION
**Pull request by filbabic**
### 🎯 Goal

Fixes #3349 

### 🛠 Implementation details

Relying on the `Dispatchers.Main.Immediate` helps us force the state/UI updates that aren't asynchronous, to be fully synchronous - i.e. to avoid dispatching events cycle.

By avoiding dispatching event cycles, the UI updates are no longer race-conditioned in Compose and things work nicely.

Lovely framework bug ^^

### 🎨 UI Changes

| Before | After |
| --- | --- |
| https://user-images.githubusercontent.com/17215808/163799261-93ef7df9-987c-4791-9a44-c68b5221b7a2.mp4 | https://user-images.githubusercontent.com/17215808/163799308-1e6d3a52-31e2-4feb-b587-1efc473afbd7.mp4 |

### 🧪 Testing

1. Run the app on develop on a live device (not emulator)
2. Try typing really fast in the message input (random bashing also works)
3. Sporadically, the message input seems to clear and reset. Also, holding the delete button (backspace) tends to stop sometimes.
4. Run the sample on this branch. Try the same thing. Typing fast should work and holding the delete button should also work.

#### General
- [x] Assigned a person / code owner group (required)
- [x] PR targets the `develop` branch
- [x] PR is linked to the GitHub issue it resolves

#### Code & documentation
- [x] Changelog is updated with client-facing changes
- [x] Comparison screenshots added for visual changes
- [x] Affected documentation updated (KDocs, docusaurus, tutorial)

### ☑️Reviewer Checklist
- [ ] UI Components sample runs & works
- [ ] Compose sample runs & works
- [ ] UI Changes correct (before & after images)
- [ ] Bugs validated (bugfixes)
- [ ] New feature tested and works
- [ ] Release notes and docs clearly describe changes
- [ ] All code we touched has new or updated KDocs

### CODE REVIEW

File: stream-chat-android-ui-common/src/main/kotlin/io/getstream/chat/android/common/composer/MessageComposerController.kt [hunk 1]
```diff
@@ -73,6 +73,10 @@ public class MessageComposerController(
     /**
      * Creates a [CoroutineScope] that allows us to cancel the ongoing work when the parent
      * ViewModel is disposed.
+     *
+     * We use the [DispatcherProvider.Immediate] variant here to make sure the UI updates don't go through the
+     * process of dispatching events. This fixes several bugs where the input state breaks when deleting or typing really
+     * fast.
      */
     private val scope = CoroutineScope(DispatcherProvider.Main)
```
Comment (skydoves) [inline stream-chat-android-ui-common/src/main/kotlin/io/getstream/chat/android/common/composer/MessageComposerController.kt:81]: Maybe should we use the `DispatcherProvider.Immediate` here?

Comment (filbabic) [reply stream-chat-android-ui-common/src/main/kotlin/io/getstream/chat/android/common/composer/MessageComposerController.kt:81]: Yeah we should thanks - I think I reverted back just to test and record the video 🤦

**Review by skydoves — APPROVED** [review]
Looks good to me! :)

**Comment by TGazica** [comment]
### ☑️Reviewer Checklist
- [x] Compose sample runs & works
- [x] UI Changes correct (before & after images)
- [x] Bugs validated (bugfixes)
- [ ] New feature tested and works
- [x] Release notes and docs clearly describe changes
- [x] All code we touched has new or updated KDocs

**Review by TGazica — APPROVED** [review]
Looks good to me as well.

**Comment by bychkovdmitry** [comment]
### ☑️Reviewer Checklist
- [ ] UI Components sample runs & works
- [x] Compose sample runs & works
- [x] UI Changes correct (before & after images)
- [x] Bugs validated (bugfixes)
- [ ] New feature tested and works
- [x] Release notes and docs clearly describe changes
- [x] All code we touched has new or updated KDocs

**Review by bychkovdmitry — APPROVED** [review]
Works great!

**Review by JcMinarro — APPROVED** [review]

**Pull request merged**